tests/integration/test_lifecycle_integration.py

```
@@ -0,0 +1,318 @@
+    resume_interrupted_runs,
+    update_inflight_checkpoint,
+)
+
+
+class _FakeRedis:
+    """In-memory Redis mock that implements the subset of commands
+    used by the lifecycle module: hset, hgetall, expire, exists,
+    delete, keys, pipeline.
+    """
+
+    def __init__(self):
```

Ruff (0.15.21)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)